src/IO/GeoJSONReader.php

```diff
@@ -0,0 +1,357 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Brick\Geo\IO;
+
+use Brick\Geo\CoordinateSystem;
+use Brick\Geo\Exception\GeometryIOException;
+use Brick\Geo\Geometry;
+use Brick\Geo\GeometryCollection;
+use Brick\Geo\LineString;
+use Brick\Geo\MultiLineString;
+use Brick\Geo\MultiPoint;
+use Brick\Geo\MultiPolygon;
+use Brick\Geo\Point;
+use Brick\Geo\Polygon;
+
+/**
+ * Builds geometries out of GeoJSON Text strings.
+ */
+class GeoJSONReader
+{
+    /**
+     * @param string $geojson The GeoJSON to read.
+     *
+     * @return GeometryCollection|Geometry
+     * @throws \Brick\Geo\Exception\CoordinateSystemException
+     * @throws \Brick\Geo\Exception\GeometryIOException
+     * @throws \Brick\Geo\Exception\InvalidGeometryException
+     * @throws \Brick\Geo\Exception\UnexpectedGeometryException
```

Comment:
My reasoning behind this is to be as specific as possible.  When writing `try{}catch($e){}` blocks, it is nice to know what exact exceptions you can catch from looking at the doc block.

Let me know your throughs on this.  I am 100% ok with making this Trow a bit more Generic if that is what you think is best.

Reply:
I usually have the same reasoning, which is why these classes exist, but then every `@throws` should document in which case the exception can be thrown.

In this specific case however, each of them ends up saying the same thing: your GeoJSON file is invalid in some way. As a user of the library, I would put a single `catch(GeometryException)` block around `read()`, I would not care whether the exception is because the JSON is malformed, a coordinate is missing, or an invalid geometry type is given: I would consider all of them an "invalid GeoJSON file".

Therefore here, I would advocate putting a single:

```
@throws GeometryException If the GeoJSON file is invalid.
```

Thoughts?

Reply:
I get your point here.  I will make the change today.  Hopefully, we can do a final review after this next batch of changes :+1:

Reply:
Resolved in commit: 3161c38